app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java

```diff
@@ -91,9 +90,17 @@
     }
 
     @Override
-    public SchemaInfo getSchemaById(int globalId) throws ArtifactNotFoundException, RegistryStorageException {
-        final ContentHandle contentHandle = storage.getArtifactByContentId(globalId);
-        return FacadeConverter.convert(contentHandle, ArtifactTypeUtil.discoverType(contentHandle, null));
+    public SchemaInfo getSchemaById(int contentId) throws ArtifactNotFoundException, RegistryStorageException {
+        final ContentHandle contentHandle = storage.getArtifactByContentId(contentId);
+        List<ArtifactMetaDataDto> artifacts = storage.getArtifactVersionsByContentId(contentId);
+        if (artifacts == null || artifacts.isEmpty()) {
+            //the contentId points to an orphaned content
+            throw new ArtifactNotFoundException("ContentId: " + contentId);
+        }
+        //get the artifact type for the first artifact version associated with it
```

Comment:
True, but if you upload a, e.g., JSON and you say it's XML you're doing that at your own risk.

Reply:
completely agree, just leaved that text for historical context